kv/dashboard.go

```diff
@@ -135,7 +135,9 @@ func filterDashboardsFn(filter influxdb.DashboardFilter) func(d *influxdb.Dashbo
 		}
 	}
 
-	return func(d *influxdb.Dashboard) bool { return true }
+	return func(d *influxdb.Dashboard) bool {
+		return ((filter.OrganizationID == nil) || (*filter.OrganizationID == d.OrganizationID))
```

Comment:
Hoping this is kosher, it follows the pattern of [labels](https://github.com/influxdata/influxdb/pull/19457/files#diff-b00232c908213b364c39e0c732ed8f1dR139)
```go
func filterLabelsFn(filter influxdb.LabelFilter) func(l *influxdb.Label) bool {
	return func(label *influxdb.Label) bool {
		return (filter.Name == "" || (strings.EqualFold(filter.Name, label.Name))) &&
			((filter.OrgID == nil) || (filter.OrgID != nil && *filter.OrgID == label.OrgID))
	}
}
```